### PR TITLE
MGMT-10722: Hostname 'localhost.localdomain' should not be valid

### PR DESCRIPTION
**Pull request by paul-maidment**
Presently the validation "hostname-valid" does nothing to ensure that
"localhost.localdomain" is not considered valid.

This PR rectifies that and introduces a unit-test to ensure that this
works.

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [x] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [x] None

## How was this code tested?

- [x] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @osherdp 
/cc @filanov 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Review by osherdp — APPROVED** [review]

**Comment by omertuc** [comment]
/hold

What about: 
image

**Comment by paul-maidment** [comment]
> /hold
> 
> What about: image

Yup, fair point, let me find an approach

File: internal/host/validator.go [hunk 1]
```diff
@@ -51,6 +51,7 @@ var (
 
 	forbiddenHostnames = []string{
 		"localhost",
+		"localhost.localdomain",
```
Comment (filanov) [inline internal/host/validator.go:54]: what about all the other hostnames form the ticket? 
```
localhost4 localhost4.localdomain4 localhost6 localhost6.localdomain6
```

Comment (filanov) [reply internal/host/validator.go:54]: just saw all the comments :)

**Comment by omertuc** [comment]
/hold

Please generate the test cases in a loop, it's a lot of duplicate code

**Review by filanov — APPROVED** [review]
i would probably suggest to simplify the tests but it's out or scope for this ticket

**Pull request merged**